src/battle/battle_command.c

```diff
@@ -5909,3 +5916,561 @@ static void *BattleScriptGetVarPointer(BattleSystem *bsys, BattleContext *ctx, i
+        }
+
+        data->state = STATE_GET_EXP_LEVEL_UP_SUMMARY_PRINT_DIFF_WAIT;
+        break;
+    }
+
+    case STATE_GET_EXP_LEVEL_UP_SUMMARY_PRINT_TRUE: {
+        TempStatsStruct monData = ov12_0226C33C;
+        Window *window = BattleSystem_GetWindow(data->bsys, 1);
+
+        FillWindowPixelRect(window, 0xF, 80, 0, 36, 96); // clear out the diff section (keep the printed stat names)
+
+        for (i = 0; i < 6; i++) {
+            msg.id = 949; // just a number
+            msg.tag = 7;
+            msg.param[0] = GetMonData(mon, monData.stats[i], NULL);
+            msg.unk1C = 3;
+
+            ov12_0223C4E8(data->bsys, window, msgLoader, &msg, 72, 16 * i, 0x2, 36, 0);
+        }
+
+        data->state = STATE_GET_EXP_LEVEL_UP_SUMMARY_PRINT_TRUE_WAIT;
+        break;
+    }
+    
+    case STATE_GET_EXP_LEVEL_UP_SUMMARY_PRINT_DIFF_WAIT:
+    case STATE_GET_EXP_LEVEL_UP_SUMMARY_PRINT_TRUE_WAIT:
+        if ((gSystem.newKeys & (PAD_BUTTON_A | PAD_BUTTON_B | PAD_BUTTON_X | PAD_BUTTON_Y)) || System_GetTouchNew()) {
+            PlaySE(SEQ_SE_DP_SELECT);
+            data->state++;
+        }
+        break;
+
+    case STATE_GET_EXP_LEVEL_UP_CLEAR:{
+        Window *window = BattleSystem_GetWindow(data->bsys, 1);
+
+        sub_0200E5D4(window, 0);
+        RemoveWindow(window);
+
+        G2_SetBG0Priority(1);
+        SetBgPriority(1, 0);
+        SetBgPriority(2, 1);
+
+        ov12_0223C224(data->bsys, 0);
+
+        if (data->ctx->selectedMonIndex[expBattler] != slot) {
+            ov12_022484D4(data->bsys, data);
+        }
+
+        FreeToHeap(data->ctx->unk_17C);
+        data->state = 15;
+        break;
+    }
+
+    case STATE_GET_EXP_CHECK_LEARN_MOVE: {
+        u16 move;
+        BgConfig *bgConfig = BattleSystem_GetBgConfig(data->bsys); // unused, but must be kept to match
+
+        switch (MonTryLearnMoveOnLevelUp(mon, &data->unk30[2], &move)) {
+        case 0:
+            data->state = STATE_GET_EXP_GAUGE;
+            break;
+        case 0xFFFE:
+            
+            break;
+        case 0xFFFF:
+            data->unk30[4] = move;
+            data->state = STATE_GET_EXP_WANTS_TO_LEARN_MOVE_PRINT;
+            break;
+        default:
+            if (data->ctx->selectedMonIndex[expBattler] == slot) {
+                BattleSystem_ReloadMonData(data->bsys,
+                    data->ctx,
+                    expBattler,
+                    data->ctx->selectedMonIndex[expBattler]);
+            }
+
+            msg.id = 4; // "{0} learned {1}!"
+            msg.tag = 10;
+            msg.param[0] = expBattler | (slot << 8);
+            msg.param[1] = move;
+            data->unk30[0] = ov12_0223C494(data->bsys, msgLoader, &msg, BattleSystem_GetTextFrameDelay(data->bsys));
+            data->state = STATE_GET_EXP_LEARNED_MOVE_WAIT;
+            break;
+        }
+
+        break;
+    }
+
+    case STATE_GET_EXP_WANTS_TO_LEARN_MOVE_PRINT:
```

Comment:
I genuinely wonder if these prints can be done in an inline/a macro, since they're very repeated, would that match?

Reply:
That would be a bad idea since while they're common the params being set are generally quite different